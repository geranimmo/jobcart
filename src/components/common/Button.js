import React from "react";

const Button = props => {
	return <button type="button" {...props}>{props.value}</button>;
};

export { Button };
