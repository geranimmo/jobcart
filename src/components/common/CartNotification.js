import React from "react";

const CartNotification = props => {
	const {
		totalItems,
		className
	} = props;
	const cartTotal = totalItems > 9 ? '9+' : totalItems;
    
	return <div className={`${className} ${totalItems === 0 ? 'hidden' : ''}`}>{cartTotal}</div>;
};

export { CartNotification };
