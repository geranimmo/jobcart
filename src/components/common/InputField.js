import React from "react";

const InputField = props => {
	return <input {...props}/>;
};

export { InputField };
