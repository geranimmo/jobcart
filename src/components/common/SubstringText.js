const SubstringText = props => {
	return props.text.length > props.textMaxLength ?
		props.text.substring(0, props.textMaxLength)+'...'
		:
		props.text;
};

export { SubstringText };