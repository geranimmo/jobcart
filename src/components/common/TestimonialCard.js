import React from 'react';

const TestimonialCard = props => {
	const { TestimonialWrapper } = styles;
    
	return (
		<div style={TestimonialWrapper} className={props.className}>
			{props.children}
		</div>
	);
};

// sample of inline style
const styles = {
	TestimonialWrapper: {
		display: 'inline-block'
	}
};

export { TestimonialCard };