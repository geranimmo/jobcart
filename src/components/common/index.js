export { InputField } from './InputField';
export { Button } from './Button';
export { CartFooter } from './CartFooter';
export { DateTimeConverter } from './DateTimeConverter';
export { SubstringText } from './SubstringText';
export { CartItem } from './CartItem';
export { CartNotification } from './CartNotification';