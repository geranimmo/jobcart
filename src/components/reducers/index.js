import { combineReducers } from 'redux';
import LoginReducer from './LoginReducers';
import ProfileReducer from './ProfileReducers';
import ShoppingCartReducer from './ShoppingCartReducers';
import PackageReducer from './PackageReducers';
import TestimonialReducer from './TestimonialReducers';
import TotalCostReducer from './TotalCostReducers';
import TotalDiscountReducer from './TotalDiscountReducers';

export default combineReducers({
	UserLogin: LoginReducer,
	Profile: ProfileReducer,
	ShoppingCart: ShoppingCartReducer,
	Packages: PackageReducer,
	Testimonials: TestimonialReducer,
	TotalCost: TotalCostReducer,
	TotalDiscount: TotalDiscountReducer
});
