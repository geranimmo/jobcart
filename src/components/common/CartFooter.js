import React from 'react';
import { Button } from './Button';

const CartFooter = props => {
	let fixedCost = parseFloat((props.totalCartCost - props.totalCartDiscount).toFixed(2));
	
	const buttonProps = {
		className: 'button__checkout',
		value: `Checkout (${props.myCartList.length})`
	};
	
	return (
		<div className={props.className}>
			<div>
				<span>
					<label>Total Cost:</label>
					${fixedCost}
				</span>
				<Button {...buttonProps}/>
			</div>
		</div>
	);
};

export { CartFooter };