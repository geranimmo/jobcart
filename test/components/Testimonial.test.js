import React from 'react';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { mount } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import TestimonialSlider from '../../src/components/testimonial';
import Slider from 'react-slick';
import { TestimonialCard } from '../../src/components/common/TestimonialCard';
import reducers from '../../src/components/reducers';
import ListTestimonials from '../../src/assets/json/testimonials__.json';

describe('>>> T E S T I M O N I A L  L I S T ---- Test & Snapshot <<<', () => {
	const store = createStore(reducers, {}, applyMiddleware(thunk));
	let mountWrapper;

	beforeEach(() => {
		mountWrapper = mount(
			<Provider store={store}>
				<Router>
					<TestimonialSlider
						testimonialList={ListTestimonials}
					/>
				</Router>
			</Provider>
		);
	});

	it('+++ Render Sliders component +++', () => {
		const finder = mountWrapper.find('.testimonial__container');
		expect(finder.length).toEqual(1);

		const sliderContentWrapper = finder.find('.testimonial__content__wrapper');
		expect(sliderContentWrapper.length).toEqual(1);
        
		const sliderContent = sliderContentWrapper.find(Slider);
		expect(sliderContent.length).toEqual(1);

		// total slider content should be 4
		const sliderCardInner = sliderContent.find(TestimonialCard);
		expect(sliderCardInner.length).toEqual(4);

		// total slider should be 1
		const sliderContents = mountWrapper.find(Slider);
		expect(sliderContents.length).toEqual(1);
	});

	it('+++ Capturing Snapshot of TestimonialSlider component +++', () => {
		const renderedValue = renderer.create(
			<Provider store={store}>
				<Router>
					<TestimonialSlider testimonialList={ListTestimonials}/>
				</Router>
			</Provider>
		).toJSON();
		
		expect(renderedValue).toMatchSnapshot();
	});
});