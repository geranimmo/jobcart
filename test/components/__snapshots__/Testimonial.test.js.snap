// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>> T E S T I M O N I A L  L I S T ---- Test & Snapshot <<< +++ Capturing Snapshot of TestimonialSlider component +++ 1`] = `
<div
  className="testimonial__container"
>
  <div
    className="testimonial__wrapper"
  >
    <div
      className="section__title"
    >
      Testimonials
    </div>
    <div
      className="testimonial__content__wrapper"
    >
      <div
        className="slick-slider slick-initialized"
        dir="ltr"
      >
        <div
          className="slick-list"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={null}
          onMouseMove={null}
          onMouseUp={[Function]}
          onTouchCancel={null}
          onTouchEnd={[Function]}
          onTouchMove={null}
          onTouchStart={[Function]}
          style={Object {}}
        >
          <div
            className="slick-track"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseOver={[Function]}
            style={
              Object {
                "WebkitTransform": "translate3d(0px, 0px, 0px)",
                "WebkitTransition": "",
                "msTransform": "translateX(0px)",
                "opacity": 1,
                "transform": "translate3d(0px, 0px, 0px)",
                "transition": "",
              }
            }
          >
            <div
              aria-hidden={false}
              className="slick-slide slick-active slick-current"
              data-index={0}
              onClick={[Function]}
              style={
                Object {
                  "outline": "none",
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <div
                  className="testimonial__list"
                  style={
                    Object {
                      "display": "inline-block",
                    }
                  }
                >
                  <div
                    className="info__user"
                  >
                    <div
                      className="user__photo"
                    >
                      <img
                        alt="Christian Eka Saputra"
                        src="/assets/images/photo/christian-eka-saputra.jpeg"
                      />
                    </div>
                    <div
                      className="user__description"
                    >
                      <span
                        className="user__name"
                      >
                        Christian Eka Saputra
                      </span>
                      <span
                        className="company__name"
                      >
                        PT. Global Visi Media
                      </span>
                    </div>
                    <div
                      className="user_testimonial"
                    >
                      <h1>
                        Classic Ad Lorem Ipsum is simply dummy.
                      </h1>
                      <h2>
                        Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.
                      </h2>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-hidden={true}
              className="slick-slide"
              data-index={1}
              onClick={[Function]}
              style={
                Object {
                  "outline": "none",
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <div
                  className="testimonial__list"
                  style={
                    Object {
                      "display": "inline-block",
                    }
                  }
                >
                  <div
                    className="info__user"
                  >
                    <div
                      className="user__photo"
                    >
                      <img
                        alt="Julien Schwarzedyan"
                        src="/assets/images/photo/julien-schwarzedyan.jpeg"
                      />
                    </div>
                    <div
                      className="user__description"
                    >
                      <span
                        className="user__name"
                      >
                        Julien Schwarzedyan
                      </span>
                      <span
                        className="company__name"
                      >
                        Valiant Store PTE. LTD.
                      </span>
                    </div>
                    <div
                      className="user_testimonial"
                    >
                      <h1>
                        Lorem Ipsum is classic Ad simply dummy.
                      </h1>
                      <h2>
                        Classic Ad simply dummy is lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.Lorem Ipsum is simply dummy text of the printing and typesetting industry.
                      </h2>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-hidden={true}
              className="slick-slide"
              data-index={2}
              onClick={[Function]}
              style={
                Object {
                  "outline": "none",
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <div
                  className="testimonial__list"
                  style={
                    Object {
                      "display": "inline-block",
                    }
                  }
                >
                  <div
                    className="info__user"
                  >
                    <div
                      className="user__photo"
                    >
                      <img
                        alt="Monari Homawash"
                        src="/assets/images/photo/monari-homawash.jpeg"
                      />
                    </div>
                    <div
                      className="user__description"
                    >
                      <span
                        className="user__name"
                      >
                        Monari Homawash
                      </span>
                      <span
                        className="company__name"
                      >
                        Gudang Televisi TBK.
                      </span>
                    </div>
                    <div
                      className="user_testimonial"
                    >
                      <h1>
                        Ad simply dummy is Lorem Ipsum classic.
                      </h1>
                      <h2>
                        The industry's of classic Ad simply dummy are lorem Ipsum and has been a standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.Lorem Ipsum is simply dummy text of the printing and typesetting industry.
                      </h2>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-hidden={true}
              className="slick-slide"
              data-index={3}
              onClick={[Function]}
              style={
                Object {
                  "outline": "none",
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <div
                  className="testimonial__list"
                  style={
                    Object {
                      "display": "inline-block",
                    }
                  }
                >
                  <div
                    className="info__user"
                  >
                    <div
                      className="user__photo"
                    >
                      <img
                        alt="Weiling Kusuma Atmadja"
                        src="/assets/images/photo/weiling-kusuma-atmadja.jpeg"
                      />
                    </div>
                    <div
                      className="user__description"
                    >
                      <span
                        className="user__name"
                      >
                        Weiling Kusuma Atmadja
                      </span>
                      <span
                        className="company__name"
                      >
                        Jual Beli Store
                      </span>
                    </div>
                    <div
                      className="user_testimonial"
                    >
                      <h1>
                        Does Lorem Ipsum classic is an ad of simply dummy.
                      </h1>
                      <h2>
                        Lorem Ipsum and has been a standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.Lorem Ipsum is simply dummy text of the printing and typesetting industry.
                      </h2>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
